File: Leetcode/Algorithm/cpp/Move Zeroes.cc
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n = nums.size();
        int p = 0;
        int q = 0;
        
        while (p < n && q < n) {
            while (q < n && nums[q] == 0) {
                q++;
            }
            if (q < n) {
                swap(nums[p], nums[q]);
                p++;
                q++;
            }
        }
    }
};

File: Leetcode/Algorithm/golang/Move Zeroes.go
func moveZeroes(nums []int)  {
    n := len(nums);
    p := 0;
    for i := 0; i < n; i++ {
        if nums[i] != 0 {
            nums[p], nums[i] = nums[i], nums[p]
            p++;
        }
    }
}
